spotirec/spotirec.py

```diff
@@ -184,14 +184,26 @@ def check_scope_permissions():
         exit(0)
 
 
-def authorize():
+def authorize(port=PORTS[0]):
     """
     Open redirect URL in browser, and host http server on localhost.
     Function index() will be routed on said http server.
     """
     logger.verbose('hosting localhost server')
-    webbrowser.open(sp_oauth.redirect)
-    run(host='', port=PORT)
+    sp_oauth.PORT = port
+    webbrowser.open(f'{sp_oauth.redirect}:{port}')
+    try:
+        logger.info(f'running authorization on {sp_oauth.redirect}:{port}')
+        run(host='', port=port, quiet=True)
+    except OSError as ex:
+        if ex.errno == 98:
```

Comment:
No handling or error-msg if `errno != 98`? I don't understand the details but maybe it's prudent to put some extra safeguarding.

Reply:
`98` is the error code for 'address in use'. I don't assume that any other codes are relevant, and if they are, they will be fixed when they appear.

Reply:
Great, just a thought